VideoClassification/readers.py

```diff
@@ -1,347 +1,200 @@
-                                   (num_segment,))
-      batch_frames = tf.reshape(tf.tile([segment_size], [num_segment]),
-                                (num_segment,))
-
-      # For segment labels, all labels are not exhausively rated. So we only
-      # evaluate the rated labels.
-
-      # Label indices for each segment, shape: [num_segment, 2].
-      label_indices = tf.stack([seg_idxs, contexts["segment_labels"].values],
-                               axis=-1)
-      label_values = contexts["segment_scores"].values
-      sparse_labels = tf.sparse.SparseTensor(label_indices, label_values,
-                                             (num_segment, self.num_classes))
-      batch_labels = tf.sparse.to_dense(sparse_labels, validate_indices=False)
-
-      sparse_label_weights = tf.sparse.SparseTensor(
-          label_indices, tf.ones_like(label_values, dtype=tf.float32),
-          (num_segment, self.num_classes))
-      batch_label_weights = tf.sparse.to_dense(sparse_label_weights,
-                                               validate_indices=False)
-    else:
-      # Process video-level labels.
-      label_indices = contexts["labels"].values
-      sparse_labels = tf.sparse.SparseTensor(
-          tf.expand_dims(label_indices, axis=-1),
-          tf.ones_like(contexts["labels"].values, dtype=tf.bool),
-          (self.num_classes,))
-      labels = tf.sparse.to_dense(sparse_labels,
-                                  default_value=False,
-                                  validate_indices=False)
-      # convert to batch format.
-      batch_video_ids = tf.expand_dims(contexts["id"], 0)
-      batch_video_matrix = tf.expand_dims(video_matrix, 0)
-      batch_labels = tf.expand_dims(labels, 0)
-      batch_frames = tf.expand_dims(num_frames, 0)
-      batch_label_weights = None
-
-    output_dict = {
-        "video_ids": batch_video_ids,
-        "video_matrix": batch_video_matrix,
-        "labels": batch_labels,
-        "num_frames": batch_frames,
-    }
-    if batch_label_weights is not None:
-      output_dict["label_weights"] = batch_label_weights
-
-    return output_dict
+	"""Truncates or pads a tensor to new_size on on a given axis.
+
+	Truncate or extend tensor such that tensor.shape[axis] == new_size. If the
+	size increases, the padding will be performed at the end, using fill_value.
+
+	Args:
+		tensor: The tensor to be resized.
+		axis: An integer representing the dimension to be sliced.
+		new_size: An integer or 0d tensor representing the new value for
+			tensor.shape[axis].
+		fill_value: Value to use to fill any new entries in the tensor. Will be cast
+			to the type of tensor.
+
+	Returns:
+		The resized tensor.
+	"""
+	tensor = tf.convert_to_tensor(tensor)
+	shape = tf.unstack(tf.shape(tensor))
+
+	pad_shape = shape[:]
+	pad_shape[axis] = tf.maximum(0, new_size - shape[axis])
+
+	shape[axis] = tf.minimum(shape[axis], new_size)
+	shape = tf.stack(shape)
+
+	resized = tf.concat([
```

Comment:
Consider tf.pad?

Reply:
I don't believe tf.pad will decrease the size if necessary